Engine/ac/viewframe.cpp

```diff
@@ -145,7 +145,11 @@ void CheckViewFrame (int view, int loop, int frame, int sound_volume) {
         }
     }
     if (sound_volume != SCR_NO_VALUE && channel != NULL)
-        channels[channel->id]->set_volume_percent(channels[channel->id]->get_volume() * sound_volume / 100);
+    {
+        AudioChannelsLock lock;
+        auto* ch = lock.GetChannel(channel->id);
+        ch->set_volume_percent(ch->get_volume() * sound_volume / 100);
```

Comment:
Worth checking if ch is null here since play_audio_clip_by_index can return null.

Reply:
In that case `if (sound_volume != SCR_NO_VALUE && channel != NULL)` condition would be false

Reply:
I mean, that's technically correct, but it's a slightly round about way of checking it. Perhaps an assert check for the benefit of future readers?